src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java

```diff
@@ -0,0 +1,457 @@
+import com.vividsolutions.jts.geom.MultiLineString;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+/**
+ * Loads an NIH shapefile into an edge-based graph, associating the edges with those previously
+ * imported from OpenStreetMap.
+ *
+ */
+public class NihShapefileStreetGraphBuilderImpl implements GraphBuilder {
+    private static Logger LOG = LoggerFactory.getLogger(ShapefileStreetGraphBuilderImpl.class);
+
+    private FeatureSourceFactory _featureSourceFactory;
+
+    private GraphServiceImpl graphService;
+
+    public List<String> provides() {
+        // streets are required by the previous OSM graph builder step, so return an empty collection
+        return Collections.emptyList();
+    }
+
+    public List<String> getPrerequisites() {
+        // require the things provided by the OSM builder
+        return Arrays.asList("streets", "turns");
+    }
+
+    public void setFeatureSourceFactory(FeatureSourceFactory factory) {
+        _featureSourceFactory = factory;
+    }
+
+    @Override
+    public void buildGraph(Graph graph, HashMap<Class<?>, Object> extra) {
+        LOG.info("Adding to graph data from NIH Shapefile...");
+
+        FeatureSource<SimpleFeatureType, SimpleFeature> featureSource = _featureSourceFactory
+                .getFeatureSource();
+
+        FeatureCollection<SimpleFeatureType, SimpleFeature> features = null;
+
+        try {
+            CoordinateReferenceSystem sourceCRS = featureSource.getInfo().getCRS();
+
+            Hints hints = new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, Boolean.TRUE);
+            CRSAuthorityFactory factory = ReferencingFactoryFinder.getCRSAuthorityFactory("EPSG",
+                    hints);
+            CoordinateReferenceSystem worldCRS = factory
+                    .createCoordinateReferenceSystem("EPSG:4326");
+
+            Query query = new Query();
+            query.setCoordinateSystem(sourceCRS);
+            query.setCoordinateSystemReproject(worldCRS);
+
+            features = featureSource.getFeatures(query);
+
+        } catch (Exception ex) {
+            throw new IllegalStateException("error loading shapefile street data", ex);
+        } finally {
+            _featureSourceFactory.cleanup();
+        }
+
+        // TODO: get rest of properties
+        StreetVertexIndexServiceImpl vertexSvc = new StreetVertexIndexServiceImpl(graph);
+        SimpleFeatureConverter<Long> osmIdSelector = new AttributeFeatureConverter("osm_id");
+        SimpleFeatureConverter<Integer> segmentIdSelector = new AttributeFeatureConverter("segment_id");
+        StringAttributeFeatureConverter nameSelector = new StringAttributeFeatureConverter("name", "");
+
+        List<SimpleFeature> featureList = new ArrayList<SimpleFeature>();
+        FeatureIterator<SimpleFeature> it2 = features.features();
+        while (it2.hasNext()) {
```

Comment:
This isn't what I'm used to seeing with iterators -- is this so that the file handle can get properly released down below? If so, it would be good to add a comment.